### DISCUSSION TITLE
Create resource account

### DISCUSSION THREAD

**Discussion by spielcrypto** [aptos-labs/aptos-developer-discussions#362] [Questions] [answered]
To create a resource account I need the SEED can it be everything? for example an email? Also I don't understand the create-resource-account-and-publish-package function, does it mean that I create a contract and after I execute that function in order to publish that contract under the address of that resources account? In that case after how can I get the signature of that resource account?

**Comment by MoonShiesty** [comment, marked as answer]
> create a resource account I need the SEED can it be everything

seed just needs to be unique per deployer account, it creates a deterministic mapping from (deployer_address, seed) => resource_account_address. so you cant publish two packages from the same deployer with the same seed

> for example an email? 

sure, thats unique

> does it mean that I create a contract and after I execute that function in order to publish that contract under the address of that resources account?

yes, that function creates a resource account, and publishes a package in that resource account

> In that case after how can I get the signature of that resource account?

the vm calls `init_module` once when the module is initialized with a signature from the modules address

you can also retrieve and store the signer_cap in `init_module` for signing later using `resource_account::retrieve_resource_account_cap`

```
    /// This function is invoked only when this package is deployed the first time.
    fun init_module(account: &signer) {
        let signer_cap = resource_account::retrieve_resource_account_cap(account, @deployer);
         move_to(account, SignerCapStore {
            signer_cap,
        });
```
